### DISCUSSION TITLE
PGbouncer TLS encryption server side using multiple CAs certs.

### DISCUSSION THREAD

**Discussion by aalbanes** [pgbouncer/pgbouncer#670] [Q&A]
Hi Team
I have pgbouncer configured to use TLS   to  connect to a remote postgres.  Postgres server is configured in  verifyCA mode  as well as  pgbouncer server .

Here a snippet of my config:
```
server_tls_sslmode = verify-ca
server_tls_ca_file = /var/server.ca_cert
server_tls_protocols = secure
server_tls_ciphers = secure
```

server.ca_cert contains the same CA cert used to sign postgres server certificate.

Everything works fine with this configuration.

Now I want to rotate my  Postgres Server certs (both server and CA certs). 
In order to minimise the service downtime,  I would like to create a temporary CA file ,containing both old and new CA certs,   to be used  on pgbouncer as "server_tls_ca_file",  so that it could be working seemless with both newer and older DB cert.

I tried this approach on a test env  but  I am getting this error when switching cert:
```
2022-01-14 11:45:35.139 UTC [19417] LOG process up: PgBouncer 1.15.0, libevent 2.1.8-stable (epoll), adns: c-ares 1.14.0, tls: OpenSSL 1.1.1  11 Sep 2018
2022-01-14 11:45:38.631 UTC [19417] WARNING C-0x55a8b4d85c80: dbtest/dbuser@127.0.0.1:55520 pooler error: pgbouncer cannot connect to server
```

I validated via openssl command if DB server cert is properly validate using "/var/server.ca_cert" CA and it was success.

```
openssl verify -verbose -CAfile /var/ccdb.server.ca_cert  server.public_cert
server.public_cert: OK
```
It seems to me that PGbouncer only uses the first cert in the CA file.
Is there a way to make multiple CA certs work at same time ?

Thx a lot for help

**Comment by petere** [comment]
The value set for `server_tls_ca_file` ends up getting passed to the OpenSSL function `SSL_CTX_load_verify_locations()`, which is the same function that core PostgreSQL uses. So there is no intentional reason why a CA file should work in one product and not in the other.
